website/project/views/node.py

```diff
@@ -586,20 +543,35 @@ def togglewatch_post(**kwargs):
         'watched': user.is_watching(node)
     }
 
+@must_be_logged_in
```

Comment:
This decorator should be unnecessary, since you already have `must_have_permission`.